### PR TITLE
Not for merging yet: Immutable buffers + persistance mapping proposal

### PR DESCRIPTION
**Pull request by mp3butcher**
Here's what I did for the moment on immutables buffer integration 
**Not candidate to merge for the moment
I pr just to make review easier**

In this implementation I made DMAbuffer a new attribute of BufferData.
The problem (if there's one) is that user should be aware to interact through this buffer and not vector interface.
I didn't find a way to use vector interface for both dma and ::new memory allocation policy without violating the standard (interact directly with vector allocator state of a boost::vector)
If this compromise is enough for community requirement I'll continue on this path (and make my PR cleaner)

### CODE REVIEW

File: src/osg/BufferObject.cpp [hunk 1]
```diff
@@ -120,7 +122,7 @@ void GLBufferObject::clear()
 void GLBufferObject::compileBuffer()
 {
     _dirty = false;
-
+_persistantDMA=0;
```
Comment (mp3butcher) [inline src/osg/BufferObject.cpp]: oups this should be removed

**Comment by filnet** [comment]
I believe "persistant" is spelled "persistent" in English.

**Comment by mp3butcher** [comment]
close/open to force travis job

**Comment by mp3butcher** [comment]
close/open to force travis job

**Comment by mp3butcher** [comment]
as i said it's an uncleaned and not deeply tested proposal
I'm waiting @robertosfield word in order to clean it an make an example

**Comment by mp3butcher** [comment]
in all case https://github.com/openscenegraph/OpenSceneGraph/pull/464/commits/d740f513b40a07dcc786e8602797ea7274820f7b is cherry pickable directly i think

**Comment by openscenegraph** [comment]
It really feels like you are trying to test me, yet again indentation is not consistent with the rest of the OSG, and you have random other fixes in there for no reason.  I really fed with having to wade through this stuff. It's not the first, second or third time I've raised this issue.  PR's to the OSG main project website are things you really need to take care over, I shouldn't have to be pushing back all the time to fix basic stuff.

As for the actual functionality attempted here. Well pushing context specific data into BufferData which is designed to be graphics context agnostic is broken, the whole point of this design is to allow the subclasses of BufferData to just concentrate on application data and not have to worry about what context are created by the application - it's all about decoupling the scene graph from the viewer.  It's one of the OSG great assets.  And here you won't to throw away a clean design for a niche feature only usable on a GL4 and has absolutely no value for all other GL/GLES versions.

With niche feature think how to implement things in a custom way where the changes are kept local, not sprayed across the core OSG.

Also the change for size_t, if this is a real issue could you post a separate PR for it and an illustrate of what build combinations it may be an issue.  If it's a genuine issue that we'll need to backport this change.

**Comment by mp3butcher** [comment]
As I mentionned everywhere this PR is dirty and not for merging...so I didn't care indentation or lexic issue for the moment...

I really would like to take pr seriously but in most of the case I must discuss first of the pr... and what better way to discuss a pr than a pr ?!

I'd though the cherrypick of https://github.com/openscenegraph/OpenSceneGraph/commit/d740f513b40a07dcc786e8602797ea7274820f7b would be obvious as size_t is a namespaceless type define in string.h (it's just names collisions paranoia talking)

Now that civilities are done let's discuss the center of indecision in this pr: 
how to implement persistent buffer in osg?
I'm agree with you on the global frame but i think you miss the fact that DMAbuffer principal usage is to write the memory in the update thread and commit changes to the renderer thread throught glcommit
It seams strange to call gl from update thread but that how it works..No?

**Comment by openscenegraph** [comment]
You are abusing the PR system for this type of discussion.  

The right place for discussions is on osg-users ML/forum, this way everyone subscribed can see, contribute and benefit the discussion rather only those who happen to follow PR threads.  I have never suggested to you that pushing all these PR's is a good thing, please stop doing it.  

As for indecision, I've got other time critical work, I'm not paid to jump immediately and review your every speculative PR.  I have to prioritize where I spend my time.  I have  even have a family that I occasionally like to spend time with too.  Time replying here is time that is better spent elsewhere.  Please stop misusing the PR system.

**Comment by mp3butcher** [comment]
okay i wil try to explain myself on the forum and put link to my branches on it... (hoping i'll not loose my time with unconcerned users..)

**Pull request closed without merging**